### DISCUSSION TITLE
EntityFrameworkEvent logging

### DISCUSSION THREAD

**Discussion by Nick-Den** [thepirat000/Audit.NET#687] [Q&A]
Hi, can i somehow add property to fired EntityFrameworkEvent?  to EventEntryChange, like overrided ColumName from annotation.
I need to store user friendly name for changes history in db. Now my best idea store dictionary with table+columnName keys and search for every changed field. But this dont sounds good)
Or mb i should go some other way?
Thank you in advance

**Comment by thepirat000** [comment]
It's difficult to determine with the limited information provided.

You can add custom fields to the EventEntry, but not to the EventEntryChange.

For example, from the audit entity action:

```c#
Audit.Core.Configuration.Setup()
    .UseEntityFramework(_ => _
        .UseDbContext<MyContext>()
        .AuditTypeMapper(t => typeof(AuditLog))
        .AuditEntityAction<AuditLog>((ev, entry, entity) =>
        {
            entry.CustomFields["Annotations"] = GetAnnotations(entry); 
        }));
```

You'll need to implement the GetAnnotations method to retrieve the necessary data for each column from the entry.

For example, you could do something like this:

```c#
private List<Annotation> GetAnnotations(EventEntry entry)
{
    return entry.Changes.ConvertAll(change =>
        new
        {
            ColumnName = change.ColumnName, 
            OverridenColumnName = GetOverridenColumnName(change.ColumnName, entry.GetEntry())
        });
}
```
